File: android/gemcore/src/main/kotlin/com/gemwallet/android/domains/search/WalletSearchConfig.kt
package com.gemwallet.android.domains.search

import uniffi.gemstone.Config

object WalletSearchConfig {
    private val config get() = Config().getWalletSearchConfig()

    val assetsInitialLimit: Int get() = config.assetsInitialLimit.toInt()

    val assetsTagLimit: Int get() = config.assetsTagLimit.toInt()

    val assetsSearchLimit: Int get() = config.assetsSearchLimit.toInt()

    val perpetualsPreviewLimit: Int get() = config.perpetualsPreviewLimit.toInt()

    val resultsLimit: Int get() = config.resultsLimit.toInt()
}

File: ios/Packages/GemstonePrimitives/Sources/WalletSearchConfig.swift
// Copyright (c). Gem Wallet. All rights reserved.

import Foundation
import class Gemstone.Config

public struct WalletSearchConfig {
    private init() {}

    public static var assetsInitialLimit: Int {
        Int(Config.shared.getWalletSearchConfig().assetsInitialLimit)
    }

    public static var assetsTagLimit: Int {
        Int(Config.shared.getWalletSearchConfig().assetsTagLimit)
    }

    public static var assetsSearchLimit: Int {
        Int(Config.shared.getWalletSearchConfig().assetsSearchLimit)
    }

    public static var perpetualsPreviewLimit: Int {
        Int(Config.shared.getWalletSearchConfig().perpetualsPreviewLimit)
    }

    public static var resultsLimit: Int {
        Int(Config.shared.getWalletSearchConfig().resultsLimit)
    }
}
